streams/src/test/java/org/apache/kafka/streams/integration/QueryableStateIntegrationTest.java

```diff
@@ -225,7 +480,73 @@ private void verifyCanGetByKey(final String[] keys,
         assertThat(countState, equalTo(expectedCount));
     }
 
-    private void waitUntilAtLeastOneRecordProcessed() throws InterruptedException {
+    /**
+     * Note: this method changes the values in expectedWindowState and expectedCount
+     * @param keys All the keys we ever expect to find
+     * @param expectedWindowState Expected window state
+     * @param expectedCount Expected cound
+     * @param windowStore Window Store
+     * @param myCount CountStore
+     * @param failIfKeyNotFound if true, tests fails if an expected key is not found in store. If false,
+     *                          the method merely inserts the new found key into the list of
+     *                          expected keys.
+     * @throws InterruptedException
+     */
+    private void verifyGreaterOrEqual(final String[] keys,
+                                      Map<String, Long> expectedWindowState,
+                                      Map<String, Long> expectedCount,
+                                      final ReadOnlyWindowStore<String, Long> windowStore,
+                                      final ReadOnlyKeyValueStore<String, Long> myCount,
+                                      boolean failIfKeyNotFound)
+        throws InterruptedException {
+        final Map<String, Long> windowState = new HashMap<>();
+        final Map<String, Long> countState = new HashMap<>();
+
+        for (String key : keys) {
+            Map<String, Long> map = fetchMap(windowStore, key);
+            if (map.equals(Collections.<String, Long>emptyMap()) && failIfKeyNotFound) {
+                fail("Key not found " + key);
+            }
+            windowState.putAll(map);
```

Comment:
It may be better to check for all windows instead of the oldest window, but I feel it is not too big deal.